web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/HsaServiceProviderImpl.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.intygstjanst.web.service.impl;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.intygstjanst.web.integration.hsa.HsaService;
+import se.inera.intyg.intygstjanst.web.service.HsaServiceProvider;
+
+@Service
+public class HsaServiceProviderImpl implements HsaServiceProvider {
```

Comment:
I think we should add another method to already existing HsaService (getHsaIdsForCareUnitAndSubUnits(...)). It can be confusing which to use... HsaServiceProvider or HsaService. They almost do the same things...

Reply:
I wrapped it into a HsaResponse and implemented function in HsaService.